### PR TITLE
Option to include astroNN orbit info (+ Galactocentric coords) when reading using apread.allStar

### PR DESCRIPTION
**Pull request by jmackereth**
Quick fix to allow for inclusion of orbit params and Galactocentric coords when matching allStar with astroNN. As written, will check for use of the DR16 catalogue - otherwise, this info isn't pulled and the original allStar is sent back.

### CODE REVIEW

File: apogee/tools/read.py [hunk 1]
```diff
@@ -584,6 +590,15 @@ def rcsample(main=False,dr=None,xmatch=None,
...
         data= _add_astroNN_ages(data,astroNNdata)
+    if use_astroNN or kwargs.get('astroNN', False) or use_astroNN_orbits:
+        _warn_astroNN_orbits()
+        astroNNdata= astroNN()
+        # Match on (ra,dec)
+        m1,m2,_= _xmatch(data,astroNNdata,maxdist=2.,
+            colRA1='RA',colDec1='DEC',colRA2='ra_apogee',colDec2='dec_apogee')
+        data= data[m1]
+        astroNNdata= astroNNdata[m2]
+        data= _add_astroNN_orbts(data,astroNNdata)
```
Comment (jobovy) [inline apogee/tools/read.py]: typo I think? Please test to make sure this works!

File: apogee/tools/read.py [hunk 2]
```diff
@@ -1337,3 +1394,6 @@ def _warn_astroNN_distances():
 
 def _warn_astroNN_ages():
     warnings.warn("Adding ages from Mackereth, Bovy, Leung, et al. (2019)")
+
+def _warn_astroNN_orbits():
+    warnings.warn("Adding orbits and Galactocentric coordinates from DR16 astroNN VAC")
```
Comment (jobovy) [inline apogee/tools/read.py]: Maybe ref. galpy papers here (general and yours)?

File: apogee/tools/read.py [hunk 3]
```diff
@@ -1265,8 +1281,13 @@ def _swap_in_astroNN(data,astroNNdata):
     return data
 
 def _add_astroNN_distances(data,astroNNDistancesdata):
+    dr= path._default_dr()
     fields_to_append= ['dist','dist_model_error','dist_error',
                        'weighted_dist','weighted_dist_error']
+    if int(dr) == 16:
+        #also have galactocentric and orbit info
+        fields_to_append= ['dist','dist_model_error','dist_error',
+                           'weighted_dist','weighted_dist_error']
```
Comment (jobovy) [inline apogee/tools/read.py]: Why did you add this? I think this code also works for DR14...

File: apogee/tools/read.py (hunk 1 above)
Comment (jmackereth) [reply apogee/tools/read.py]: this was indeed a typo, but there was also some issues with this which are now fixed

File: apogee/tools/read.py (hunk 2 above)
Comment (jmackereth) [reply apogee/tools/read.py]: done!

File: apogee/tools/read.py (hunk 3 above)
Comment (jmackereth) [reply apogee/tools/read.py]: Sorry, this got left in from a previous version I was using to test things - removed in the new version

File: apogee/tools/read.py [hunk 4]
```diff
@@ -571,19 +583,43 @@ def rcsample(main=False,dr=None,xmatch=None,
...
+            data = _add_astroNN_ages(data,astroNNdata)
+        else:
+            astroNNdata = astroNN() #ages are in the main VAC for DR > 16
+            m1,m2,_= _xmatch(data,astroNNdata,maxdist=2.,
+                colRA1='RA',colDec1='DEC',
+                             colRA2='ra_apogee',
+                             colDec2='dec_apogee')
+            data= data[m1]
+            astroNNdata= astroNNdata[m2]
+            data= _add_astroNN_ages(data,astroNNdata)
```
[the remark this answers is not in the document]
Comment (jmackereth) [reply apogee/tools/read.py]: Hi Jo, I did this because when testing I realised that the DR14 Ages catalogue has no RA/DEC, so the match as it was didn't work (at least for me). The ID based match that `_add_astroNN_ages` does is ok in this case, I think? I may be wrong, but this seemed to work when I tested it.

for DR16, the match as before works fine because it uses the full VAC (as you mentioned).

[the remark this answers is not in the document]
Comment (jobovy) [reply apogee/tools/read.py]: I see, okay. But is it necessary to do the ``xmatch`` at all then, if there is an ID-based match after? I know it's a detail...

[the remark this answers is not in the document]
Comment (jmackereth) [reply apogee/tools/read.py]: ah I see what you mean, yes, that is pretty dumb of me.... will switch that up now

[the remark this answers is not in the document]
Comment (jmackereth) [reply apogee/tools/read.py]: ah - no I remember now: in the case of DR16 `_add_astroNN_ages` just assumes a line-by-line match, so we need the xmatch by RA and Dec for the RC sample after loading with `astroNN()`
